### DISCUSSION TITLE
Rollup packaging error, how to handle it

### DISCUSSION THREAD

**Discussion by parlay96** [rollup/plugins#1552] [Q&A]
![b1a5a1711a9dea0cb97e10079b51139f_258008980-2beead2e-972d-4f68-9800-102d5b2147d5](https://github.com/rollup/plugins/assets/32978177/3fcd037a-70f8-402c-a5ee-40b1f015cc8c)

**Comment by parlay96** [comment]
I tried using a third-party package, but there is a TS file in it, which caused me to have a problem with the packaging. I don't know how to handle it

**Comment by parlay96** [comment]
Why would it introduce nodes by default_ The ts file in modules

**Comment by TrickyPi** [comment]
It would be great if you could provide a minimal reproduction.

**Comment by parlay96** [comment]
@TrickyPi https://github.com/livelyPeng/rollup-demo,  This is a small demo. Could you please help me take a look? Thank you

**Comment by parlay96** [comment]
![image](https://github.com/rollup/plugins/assets/32978177/7f26dd7d-3d92-4d59-88f3-0399cb9864ab)
When I delete this TS file and roll it up again, it will be successful

**Comment by parlay96** [comment]
I am very grateful for your help in addressing this extreme issue @TrickyPi

**Comment by parlay96** [comment]
@TrickyPi Do you have an answer, hee hee hee

**Comment by TrickyPi** [comment]
I debugged it for a while, the reason of resolving a `.ts` file is that the old version of `@rollup/plugin-typescript` cannot filter files on `resolveId` hook. The following steps can fix this problem.
1. upgrade `@rollup/plugin-typescript` to the latest.
2. config the `include` option such as `src/*.ts` on `@rollup/plugin-typescript`.

**Comment by parlay96** [comment]
@TrickyPi @Rollup/plugin type script upgrade, does Rollup need to be upgraded?

**Reply by TrickyPi** [reply]
Unnecessary. But upgrading Rollup also is Ok.

**Comment by parlay96** [comment]
Ok, thank you very much, Momoda

**Comment by parlay96** [comment]
I think this is a problem worth noting because others may also encounter it

**Reply by TrickyPi** [reply]
I don't think it's necessary because of the `11.0.0` version of `@rollup/plugin-typescript` has fixed this problem.

**Comment by parlay96** [comment]
哟西，You're amazing
